### ISSUE TITLE
Worker failed to boot. Worker was terminated due to signal 15

### ISSUE THREAD

**Issue by gtufaile** [benoitc/gunicorn#2755]
Hello guys, Im pretty new using gunicorn (django+aws ubuntu), and Im really struggling to find out which is the source of my problem. Below there is my gunicorn log.

[2022-03-07 17:47:03 +0000] [396998] [DEBUG] Current configuration:
  config: config/gunicorn/dev.py
  wsgi_app: config.wsgi:application
  bind: ['0.0.0.0:8000']
  backlog: 2048
  workers: 2
  worker_class: sync
  threads: 1
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: True
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  print_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /home/ubuntu/contacomigo_backend
  daemon: True
  raw_env: []
  pidfile: /var/run/gunicorn/dev.pid
  worker_tmp_dir: None
  user: 1000
  group: 1000
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: /var/log/gunicorn/dev.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: /var/log/gunicorn/dev.log
  loglevel: debug
  capture_output: True
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  syslog_addr: udp://localhost:514
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags:
    statsd_prefix:
  proc_name: None
  default_proc_name: gunicorn
  pythonpath: None
  paste: None
  on_starting: <function OnStarting.on_starting at 0x7f0432b651f0>
  on_reload: <function OnReload.on_reload at 0x7f0432b65310>
  when_ready: <function WhenReady.when_ready at 0x7f0432b65430>
  pre_fork: <function Prefork.pre_fork at 0x7f0432b65550>
  post_fork: <function Postfork.post_fork at 0x7f0432b65670>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x7f0432b65790>
  worker_int: <function WorkerInt.worker_int at 0x7f0432b658b0>
  worker_abort: <function WorkerAbort.worker_abort at 0x7f0432b659d0>
  pre_exec: <function PreExec.pre_exec at 0x7f0432b65af0>
  pre_request: <function PreRequest.pre_request at 0x7f0432b65c10>
  post_request: <function PostRequest.post_request at 0x7f0432b65ca0>
  child_exit: <function ChildExit.child_exit at 0x7f0432b65dc0>
  worker_exit: <function WorkerExit.worker_exit at 0x7f0432b65ee0>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x7f0432b7a040>
  on_exit: <function OnExit.on_exit at 0x7f0432b7a160>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2022-03-07 17:47:03 +0000] [396998] [INFO] Starting gunicorn 20.1.0
[2022-03-07 17:47:03 +0000] [396998] [DEBUG] Arbiter booted
[2022-03-07 17:47:03 +0000] [396998] [INFO] Listening at: http://0.0.0.0:8000 (396998)
[2022-03-07 17:47:03 +0000] [396998] [INFO] Using worker: sync
[2022-03-07 17:47:03 +0000] [397000] [INFO] Booting worker with pid: 397000
[2022-03-07 17:47:03 +0000] [397001] [INFO] Booting worker with pid: 397001
[2022-03-07 17:47:03 +0000] [396998] [DEBUG] 2 workers
[2022-03-07 17:47:05 +0000] [396998] [WARNING] Worker with pid 397001 was terminated due to signal 15
[2022-03-07 17:47:05 +0000] [396998] [INFO] Shutting down: Master
[2022-03-07 17:47:05 +0000] [396998] [INFO] Reason: Worker failed to boot.

**Comment by thebeardedpapi** [comment]
I think signal 15 means that the app didn't start properly

**Comment by gtufaile** [comment]
Yeah, but do you know how can I get a more specific log to find out what is causing my app to not start properly?

**Comment by thebeardedpapi** [comment]
not really, can you post the gunicorn command that you're running?

```
gunicorn -w 3 -b "0.0.0.0:8000" path.to.app:app
```

if it looks something like this, i'd start with your app.py file. make sure that you have something like 
```
def create_app():
    app = Flask(__name__)
    return app

app = create_app()
```

**Comment by gtufaile** [comment]
Just figure out how to find the problem. Instead of running the application through gunicorn, I started the app using the runserver command. The problem popped in my face. Sometimes we just have to get away for a while and then comeback with these simple workarounds. Basically the problem was a misinstalled library at aws. Solved correcting the installation and now im will cofigure docker to not have these kind of problems again.
